src/lxc/log.c

```diff
@@ -170,10 +170,13 @@ static int log_append_logfile(const struct lxc_log_appender *appender,
 		     event->locinfo->file, event->locinfo->func,
 		     event->locinfo->line);
 
-	n += vsnprintf(buffer + n, sizeof(buffer) - n, event->fmt,
-		       *event->vap);
+	if (n < 0)
+		return n;
 
-	if (n >= sizeof(buffer) - 1) {
+	if (n < sizeof(buffer) - 1)
```

Comment:
Note that the synopsis of snprintf():
int snprintf(char *str, size_t size, const char *format, ...);

If n is bigger than sizeof(buffer), sizeof(buffer) - n is still positive. This is why we cannot call vsnprintf() unconditionally.